src/Option/Option.ts

```diff
@@ -142,6 +142,29 @@ export const map = <A extends NonNullish, B extends NonNullish>(
         none: none,
     })
 
+/**
+ * Execute an arbitrary side-effect function if `Some`.
+ * Does not call the function for `None`.
+ *
+ * @group Utils
+ *
+ * @example
+ * pipe(
+ *     Option.some("cheese"),
+ *     Option.iter(s => fooCallback(s))
+ * ) // fooCallback is called with "cheese"
+ *
+ * pipe(
+ *     Option.none,
+ *     Option.iter((s: string) => fooCallback(s))
+ * ) // fooCallback is not called
+ */
+export const iter = <A extends NonNullish>(f: (a: A) => void) =>
+    match<A, void>({
+        some: a => f(a),
+        none: () => {},
+    })
```

Comment:
does this work?

```suggestion
    match<A, void>({
        some: f,
        none: void 0,
    })
```

Reply:
Beautiful 😍 I tried `void` but was missing the `0`.

Reply:
Updated!